src/dsp_tools/utils/xmlupload/resource_create_client.py

```diff
@@ -185,88 +199,55 @@ def _make_color_value(value: XMLValue) -> dict[str, Any]:
     }
 
 
-def _make_date_value(value: XMLValue) -> dict[str, Any]:
-    entire_pattern = r"""
-    ^
-    (?:(GREGORIAN|JULIAN|ISLAMIC):)?        # optional calendar
-    (?:(CE|BCE|BC|AD):)?                    # optional era
-    (\d{4}(?:-\d{1,2})?(?:-\d{1,2})?)       # date
-    (?::(CE|BCE|BC|AD))?                    # optional era
-    (?::(\d{4}(?:-\d{1,2})?(?:-\d{1,2})?))? # optional date
-    $
-"""
-    assert isinstance(value.value, str)
-    date_match = regex.search(entire_pattern, value.value, flags=regex.VERBOSE)
-    assert date_match
-    date_groups = date_match.groups()
-    match date_groups:
-        case (str() | None, str() | None, str(), str() | None, str() | None):
-            # start_era cannot be None, and this provides a useful type hint
-            pass
-        case ("ISLAMIC", era, _, end_era, _) if era or end_era:
-            raise UserError(f"ISLAMIC calendar does not support eras: {value.value}")
+def _validate_boolean(s: str) -> bool:
```

Comment:
perhaps rather `_cast_to_boolean()`?

Reply:
I was actually unsure about the naming of this one. `cast` to me implies that no actual change happens, so I now simply named it `_to_boolean()`, because at the end of the day, it just takes a string and turns it into a boolean. But if you like `_cast_to_boolean()` better, then we can always change it, I don't feel strongly about this one.